### PR TITLE
Rename GenesisTimeProvider to GenesisDataProvider and add genesis validators root

### PR DESCRIPTION
**Pull request by rolfyone**
Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: validator/beaconnode/src/main/java/tech/pegasys/teku/validator/beaconnode/TimeBasedEventAdapter.java [hunk 1]
```diff
@@ -87,7 +88,7 @@ private boolean isTooLate(final UInt64 scheduledTime, final UInt64 actualTime) {
   public SafeFuture<Void> start() {
     // Don't wait for the genesis time to be available before considering startup complete
     // The beacon node may not be available or genesis may not yet be known.
-    genesisTimeProvider.getGenesisTime().thenAccept(this::start).reportExceptions();
+    genesisDataProvider.getGenesisData().thenAccept(this::start).reportExceptions();
```
Comment (ajsutton) [inline validator/beaconnode/src/main/java/tech/pegasys/teku/validator/beaconnode/TimeBasedEventAdapter.java]: Could this continue using `genesisDataProvider.getGenesisTime()`?

File: validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java [hunk 2]
```diff
@@ -23,79 +23,107 @@
...
 
     // Retry is scheduled.
     asyncRunner.executeQueuedActions();
-    verify(validatorApiChannel, times(2)).getGenesisTime();
+    verify(validatorApiChannel, times(2)).getGenesisData();
     assertThat(result).isCompletedWithValue(GENESIS_TIME);
   }
+
+  @Test
+  void shouldRetryWhenGenesisValidatorsRootIsNotYetKnown() {
```
Comment (ajsutton) [inline validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java]: This case is the same as `shouldRetryWhenGenesisTimeIsNotYetKnown`

**Review by ajsutton — COMMENTED** [review]
This looks like it's making the genesis validator root available from `GenesisDataProvider` but `ForkProvider` is still using the old Api and getting the validator root from there.

This is a good first step and could be committed as-is but probably should update the PR title and description to match what it does.

File: validator/beaconnode/src/main/java/tech/pegasys/teku/validator/beaconnode/TimeBasedEventAdapter.java (hunk 1 above)
Comment (rolfyone) [reply validator/beaconnode/src/main/java/tech/pegasys/teku/validator/beaconnode/TimeBasedEventAdapter.java]: yep it got caught in a rename

File: validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java (hunk 2 above)
Comment (rolfyone) [reply validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java]: yep that's what i copied, i wanted to show that validatorsRoot would work, i'll think of something trivial

Comment (ajsutton) [reply validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java]: But both are just testing `getGenesisData`.  I don't think we need to individually check that the shortcut `getGenesisTime` and `getGenesisValidatorRoot` methods work - they are a trivial map from `getGenesisData`.

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**